src/plugins/intel_npu/cmake/features.cmake

```diff
@@ -4,29 +4,10 @@
 
 ov_option(ENABLE_MLIR_COMPILER "Enable compilation of npu_mlir_compiler libraries" ON)
 
-ov_option(ENABLE_NPU_RUNTIME_COMMON "Enable compilation of npu runtime common libraries" ON)
+ov_option(ENABLE_NPU_PLUGIN_ENGINE "Enable compilation of NPU plugin engine" ON)
```

Comment:
This was added in prior PR, but we have ENABLE_INTEL_NPU and ENABLE_INTEL_NPU_INTERNAL.
Why we need another option?

Reply:
We keep only one option right now. In the case of the CiD build we need only the AL folder to be enabled. We don't need the rest of them.

Right now these folders are dependent on each other so add them to the same option.

Reply:
In CiD job we build selected target so no option should be necessary. It's probably ov_dev_targets which are unnecessarily included.

Reply:
If default option is ON, then the targets will be part of openvino::runtime in static build, and we have to close the option manually, FYI

Reply:
Is this desired to link plugin's code along openvino::runtime target? I would expect we rather  have separate link target for that code.